core/src/main/java/org/apache/calcite/avatica/remote/SingleBearerTokenProvider.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.avatica.remote;
+
+import org.apache.calcite.avatica.BuiltInConnectionProperty;
+import org.apache.calcite.avatica.ConnectionConfig;
+
+import java.io.IOException;
+
+public class SingleBearerTokenProvider implements BearerTokenProvider {
```

Comment:
That name sounds off. Every provider provides a single token (at a time).
Wouldn't SimpleBearerTokenProvider be more descriptive ?

Reply:
This object can provide a single token for its entire lifetime. Maybe Constant?
I'd avoid "Simple" as it may sound that the token is simple in some way.

Reply:
Constant is better.